### DISCUSSION TITLE
Save grid state in localstorage and restore

### DISCUSSION THREAD

**Discussion by Yadav0141** [generic-ui/generic-ui#45] [Q&A] [answered]
Hi There,

We wanted to use this nice angular grid library , but we also wanted to save grid state on each user basis to DB or localstorage.Is their any way to achive this?

**Comment by generic-ui** [comment, marked as answer]
Hello @Yadav0141 ,

I think that you can observe changes on grid outputs and then when something changes store it in ls or db. The list of all of the outputs is here https://generic-ui.com/guide/options

**Reply by Yadav0141** [reply]
We want to store the following things to be stored in DB for a user:-

1) ColumnConfig(Which are hidden, Not-Hidden Both).
2) Sorting applied on all columns(ASC, DESC).
3) Ordering of columns in the grid.
4) Filters applied.

Can you please elaborate more on which Output events I can observe to get values to prepare my custom model and store in DB as JSON and after refresh restores back in UI?. It would be very helpful. Thanks, Buddy

**Reply by generic-ui** [reply]
Thanks for precise list of things that you need. Right now grid doesn't provide all of requested by you information, but we will address it in the next release. It is a small change it should not take long.

**Reply by Yadav0141** [reply]
Thanks, Please once drop the comment once released.
